src/laser/Laser.cpp

```diff
@@ -0,0 +1,104 @@
+#include "Laser.H"
+#include "utils/Constants.H"
+#include "Hipace.H"
+#include "utils/HipaceProfilerWrapper.H"
+
+void
+Laser::ReadParameters ()
+{
+    amrex::ParmParse pp(m_name);
+    m_use_laser = queryWithParser(pp, "a0", m_a0);
+    if (!m_use_laser) return;
+    amrex::Vector<amrex::Real> tmp_vector;
+    if (queryWithParser(pp, "w0", tmp_vector)){
+        AMREX_ALWAYS_ASSERT_WITH_MESSAGE(tmp_vector.size() == 2,
+        "The laser waist w0 must be provided in x and y, "
+        "so laser.w0 should contain 2 values");
+        for (int i=0; i<2; i++) m_w0[i] = tmp_vector[i];
+    }
+
+    bool length_is_specified = queryWithParser(pp, "L0", m_L0);;
+    bool duration_is_specified = queryWithParser(pp, "tau", m_tau);
+    AMREX_ALWAYS_ASSERT_WITH_MESSAGE( length_is_specified + duration_is_specified == 1,
+        "Please specify exlusively either the pulse length L0 or the duration tau of the laser");
+    if (duration_is_specified) m_L0 = m_tau/get_phys_const().c;
+    queryWithParser(pp, "lambda0", m_lambda0);
+    amrex::Array<amrex::Real, AMREX_SPACEDIM> loc_array;
+    queryWithParser(pp, "position_mean", loc_array);
+    for (int idim=0; idim < AMREX_SPACEDIM; ++idim) m_position_mean[idim] = loc_array[idim];
+
+}
+
+
+void
+Laser::InitData (const amrex::BoxArray& slice_ba,
+                 const amrex::DistributionMapping& slice_dm)
+{
+    if (!m_use_laser) return;
+    HIPACE_PROFILE("Laser::InitData()");
+    // Alloc 2D slices
+    // Need at least 1 guard cell transversally for transverse derivative
+    int nguards_xy = std::max(1, Hipace::m_depos_order_xy);
+    m_slices_nguards = {nguards_xy, nguards_xy, 0};
+AMREX_ALWAYS_ASSERT(WhichLaserSlice::N == m_nslices);
+    for (int islice=0; islice<WhichLaserSlice::N; islice++) {
+        m_slices[islice].define(
+            slice_ba, slice_dm, 1, m_slices_nguards, // prev Comps[islice]["N"] instead of 1
```

Comment:
```suggestion
            slice_ba, slice_dm, 1, m_slices_nguards, /* prev Comps[islice]["N"] instead of 1 */
```
I think we observed issues with some compilers when putting such comments in the list of function arguments.